meshmode/mesh/refinement/__init__.py

```diff
@@ -20,13 +20,14 @@
 THE SOFTWARE.
 """
 
+import itertools
+from functools import partial
 
 import numpy as np
-import itertools
-from pytools import RecordWithoutPickling
-from meshmode.mesh.refinement.no_adjacency import (  # noqa: F401
-        RefinerWithoutAdjacency)
```

Comment:
I wanted to push this diff, effectively deprecating this refiner, but I found I didn't have permission. Could you do that?
```diff
diff --git a/meshmode/mesh/refinement/__init__.py b/meshmode/mesh/refinement/__init__.py
index 08c1c6b..a9ed805 100644
--- a/meshmode/mesh/refinement/__init__.py
+++ b/meshmode/mesh/refinement/__init__.py
@@ -123,6 +123,10 @@ class Refiner:
     # {{{ constructor
 
     def __init__(self, mesh):
+        from warnings import warn
+        warn("Refiner is deprecated and will be removed in 2022.",
+                DeprecationWarning, stacklevel=2)
+
         if mesh.is_conforming is not True:
             raise ValueError("Refiner can only be used with meshes that are known "
                     "to be conforming. If you would like to refine non-conforming "
@@ -974,9 +978,13 @@ class Refiner:
 
 def refine_uniformly(mesh, iterations, with_adjacency=False):
     if with_adjacency:
-        refiner = Refiner(mesh)
-    else:
-        refiner = RefinerWithoutAdjacency(mesh)
+        # For conforming meshes, even RefinerWithoutAdjacency will reconstruct
+        # adjacency from vertex identity.
+
+        if not mesh.is_conforming:
+            raise ValueError("mesh must be conforming if adjacency is desired")
+
+    refiner = RefinerWithoutAdjacency(mesh)
 
     for _ in range(iterations):
         refiner.refine_uniformly()
```

Reply:
Annoyingly, there doesn't seem to be a way to allow access just on this PR. 

I'll apply that! Thanks!

Reply:
d6f13cc